File: SensateIoT.Platform/SensateIoT.Platform.Network/SensateIoT.Platform.Network.Data/DTO/Command.cs
/*
 * Async command model.
 *
 * @author Michel Megens
 * @email  michel@michelmegens.net
 */

using SensateIoT.Platform.Network.Data.Enums;

namespace SensateIoT.Platform.Network.Data.DTO
{
	public class Command
	{
		public CommandType Cmd { get; set; }
		public string Arguments { get; set; }
	}
}

File: SensateIoT.Platform.Network.LiveDataService/src/commands/command.ts
/*
 * MQTT command interface/model.
 * 
 * @author Michel Megens
 * @email  michel@michelmegens.net
 */

export interface Command<T> {
    cmd: string;
    arguments: T;
}

interface InternalCommand {
    cmd: string;
    arguments: string;
}

export function stringifyCommand<T>(cmd: Command<T>): string {
    const args = JSON.stringify(cmd.arguments);
    const internal: InternalCommand = {
        arguments: args,
        cmd: cmd.cmd
    };

    return JSON.stringify(internal);
}
